app/lib/foreman/http_proxy.rb

```diff
@@ -51,7 +70,9 @@ def foreman_logger
     end
 
     def log_proxied_request(lib, current_proxy, requested_host)
-      foreman_logger.info "(#{lib}) Proxying request to #{requested_host} via #{current_proxy}"
+      parsed_url = URI(current_proxy)
+      parsed_url.password = '****' if parsed_url.password.present?
+      foreman_logger.info "(#{lib}) Proxying request to #{requested_host} via #{parsed_url}"
```

Comment:
Can we please have this as a separate change so we can cherry pick this to stable branches? Though I'd prefer a block form to avoid the overhead if the logging level is above `info`? Untested, but I think this should work:
```suggestion
      foreman_logger.info do
        parsed_url = URI(current_proxy)
        parsed_url.password = '****' if parsed_url.password.present?
        "(#{lib}) Proxying request to #{requested_host} via #{parsed_url}"
      end
```